docs/design/proposals/codegen-refactor/proposal-summary.md

```diff
@@ -0,0 +1,57 @@
+### Key Terms
+* `ackgenconfig`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/generate/config/config.go#L24) representation of *generator.yaml*. an **input** to *code-generator*.
+* `resource` | `k8s-resource` | `ackcrd`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/crd.go#L63) reprensenting a single top-level resource in an AWS Service. *code-generator* generates these resources using heuristics and `ackgenconfig`.
+* `shape` | `aws-sdk` | `sdk-shape` | `sdk`: the original operations, models, errors, structs for a given AWS service. sourced from *aws-sdk*, ex: [aws-sdk-go s3](https://github.com/aws/aws-sdk-go/blob/main/models/apis/s3/2006-03-01/api-2.json#L1)
+* `reconciliation`: logic involving access to or relations between `resource`, `shape`, `ackgenconfig`, and `aws-sdk`
+* `ackmodel`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/model.go#L36) representation of ACK's view of the world; the source of truth for `aws-sdk`, `ackgenconfig`, and `reconciliation`
+
+
+## Problem
+*code-generator's* accelerated growth is resulting in tech debt that degrades the experience for platform maintainers and contributors. The codebase is becoming difficult to read and extend given the quick addition of features to onboard more services. This doc summarizes and introduces the proposals to eliminate significant tech debt so that development on ACK Platform is as pleasant and clear as it is open.
+
+
+## Solution
+
+### Overview
+To improve the implementation, I propose consolidating `ackconfig` access and any *ackconfig-aws-sdk* `reconciliation` into `ackmodel`. Taking this idea further, `ackmodel` should be the "source of truth"/required input for `api` and `controller` commands instead of *generator.yaml* and *aws-sdk*.
```

Comment:
++ I like the idea of a single-pane of glass with `ackmodel`! Seems much easier to reason about.

Reply:
By combining config with ACK model, we are able to create this "source of truth" so that downstream commands (apis, controller) don't need to "think" for themselves, but instead focus on single responsibilities like how to generate Go code. For example, `SetResource` is accessing and applying `ackgenconfig` as well as generating Go code which has overloaded it with responsibility. Also, this work of applying `ackgenconfig` and resolving it with `aws-sdk` should have already been done since it would be required to generate the CRDs before kicking this process off; therefore, removing duplicate work.

We also organize the codebase significantly by consolidating this logic to a central module and extending code-generator becomes more intuitive. If I can't find a field, resolve an api inference, or need info about `ackgenconfig`, `aws-sdk`, etc. then ask the source of truth, `ackmodel`. Compared to today's less clear experience where you can access `ackgenconfig` in a number of ways (directly or thru CRD) and resolve yourself or try to find a related helper function. It'd also easier to extend config/sdk-resolve logic as it will all live `model` pkg vs. spread out like today, `code/common.go`, `model/crd.go` for example.